frontend/src/components/VisitRouter/index.js

```diff
@@ -38,13 +40,33 @@ const VisitRouter = observer(() => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const handleSubmitVisit = () => {
+  const handleSubmitVisit = async () => {
     // TODO: logic protecting against concurrent queue entries
     // if existing visit we are coming from QueueTable, so update visit
-    if (existingVisit.id) {
-      participantStore.updateVisit()
-    } else {
-      participantStore.createVisit()
+    try {
+      let isValid = await validateVisitForm(existingVisit)
+      if (existingVisit.id) {
+        !isValid.length
+          ? participantStore.updateVisit()
+          : isValid.map(error =>
```

Comment:
just curious about what happens here with the multiple errors. Does it display them one after the other? how do they come up in sequence?

Reply:
This is a good catch. I need to fix this. It's not displaying multiple errors.

Reply:
Or well, if you fix the one error and attempt a resubmit it will still display for the next error.